lib/features/pfis/pfi_page.dart

```diff
@@ -14,139 +16,182 @@ class PfiPage extends HookConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final selectedPfi = useState<Pfi?>(null);
 
-    useEffect(
-      () {
-        Future.delayed(
-          Duration.zero,
-          () => ref.read(pfisProvider.notifier).reload(),
-        );
-        return null;
-      },
-      [],
-    );
+    useEffect(() {
+      Future.delayed(
+          Duration.zero, () => ref.read(pfisProvider.notifier).reload());
+      return null;
+    }, []);
 
     return Scaffold(
       appBar: AppBar(),
       body: SafeArea(
         child: Column(
           crossAxisAlignment: CrossAxisAlignment.stretch,
           children: [
-            _buildHeader(
-              context,
-              Loc.of(context).getStartedWithAPfi,
-              Loc.of(context).selectAPfi,
-            ),
+            _buildHeader(context, ref, Loc.of(context).getStartedWithAPfi,
+                Loc.of(context).selectAPfi),
             Expanded(
-              child: _buildPfiList(context, ref, selectedPfi),
+              child: _buildList(context, ref, selectedPfi),
             ),
             const SizedBox(height: Grid.xs),
-            _buildNextButton(context, ref, selectedPfi.value),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: Grid.side),
+              child: FilledButton(
+                onPressed: selectedPfi.value == null
+                    ? null
+                    : () {
+                        Navigator.of(context).push(
+                          MaterialPageRoute(
+                            builder: (_) => AgreementPage(
+                              pfi: selectedPfi.value!,
+                            ),
+                          ),
+                        );
+                      },
+                child: Text(Loc.of(context).next),
+              ),
+            ),
           ],
         ),
       ),
     );
   }
 
-  Widget _buildHeader(BuildContext context, String title, String subtitle) =>
-      Padding(
-        padding: const EdgeInsets.symmetric(
-          horizontal: Grid.side,
-          vertical: Grid.xs,
-        ),
-        child: Column(
-          children: [
-            Align(
-              alignment: Alignment.topLeft,
-              child: Text(
-                title,
-                style: Theme.of(context).textTheme.headlineSmall?.copyWith(
-                      fontWeight: FontWeight.bold,
-                    ),
-              ),
+  Widget _buildHeader(
+      BuildContext context, WidgetRef ref, String title, String subtitle) {
```

Comment:
we're using [arrow syntax](https://dart.dev/language/functions) whenever possible now, but this looks like you just need to pull in the latest changes!